sdk/webpubsub/azure-webpubsub-client/azure/webpubsub/client/_client.py

```diff
@@ -0,0 +1,835 @@
+_RETRY_BACKOFF_MAX = 120.0
+_RECOVERY_TIMEOUT = 30.0
+_RECOVERY_RETRY_INTERVAL = 1.0
+_USER_AGENT = "User-Agent"
+
+
+class WebPubSubClient:  # pylint: disable=client-accepts-api-version-keyword,too-many-instance-attributes
+    """WebPubSubClient
+
+    :param credential: The url to connect or credential to use when connecting. Required.
+    :type credential: str or WebPubSubClientCredential
+    :keyword bool auto_rejoin_groups: Whether to enable restoring group after reconnecting
+    :keyword azure.webpubsub.client.WebPubSubProtocolType protocol_type: Subprotocol type
+    :keyword int reconnect_retry_total: total number of retries to allow for reconnect. If 0, it means disable
+     reconnect. Default is 3.
+    :keyword float reconnect_retry_backoff_factor: A backoff factor to apply between attempts after the second try
+     (most errors are resolved immediately by a second try without a delay). In fixed mode, retry policy will always
+     sleep for {backoff factor}. In 'exponential' mode, retry policy will sleep for:
+     "{backoff factor} * (2 ** ({number of retries} - 1))" seconds. If the backoff_factor is 0.1, then the
+     retry will sleep for [0.0s, 0.2s, 0.4s, ...] between retries. The default value is 1.0.
+    :keyword float reconnect_retry_backoff_max: The maximum back off time. Default value is 120.0 seconds
+    :keyword RetryMode reconnect_retry_mode: Fixed or exponential delay between attemps, default is exponential.
+    :keyword int message_retry_total: total number of retries to allow for sending message. Default is 3.
+    :keyword float message_retry_backoff_factor: A backoff factor to apply between attempts after the second try
+     (most errors are resolved immediately by a second try without a delay). In fixed mode, retry policy will always
+     sleep for {backoff factor}. In 'exponential' mode, retry policy will sleep for:
+     "{backoff factor} * (2 ** ({number of retries} - 1))" seconds. If the backoff_factor is 0.1, then the
+     retry will sleep for [0.0s, 0.2s, 0.4s, ...] between retries. The default value is 1.0.
+    :keyword float message_retry_backoff_max: The maximum back off time. Default value is 120.0 seconds
+    :keyword RetryMode message_retry_mode: Fixed or exponential delay between attemps, default is exponential.
+    :keyword bool auto_rejoin_groups: auto_rejoin_groups, default is True
+    :keyword str user_agent: The user agent to be used for the request. If specified, this will be added in front of
+     the default user agent string.
+    :keyword bool logging_enable: Whether to output network trace logs to the logger. Default is `False`.
+    :keyword float ack_timeout: Time limit to wait for ack message from server. The default value is 30.0 seconds.
+    :keyword float start_timeout: Time limit to wait for successful client start. The default value is 30.0 seconds.
+    """
+
+    def __init__(
+        self,
+        credential: Union[WebPubSubClientCredential, str],
```

Comment:
We do not allow `str` as a credential since a) it cannot roll/be updated dynamically and b) is generally very ambiguous w. regards to what it actually is. Is it a connection string? A token? An API key? Something else? And new auth methods get added over time, so even if it isn't unclear now, it likely will become an issue in the future.

Reply:
It is connection string which is declared in [readme.md](https://github.com/Azure/azure-sdk-for-python/tree/f392fb5871d95546c16ea6ac399e93d786036582/sdk/webpubsub/azure-webpubsub-client#2-connect-with-your-web-pubsub-resource). If users just want to open connection in short time, it is convenient to use the string directly since they can get it in Azure Portal. As for the expired problem, I declare it in docstring: 
https://github.com/Azure/azure-sdk-for-python/blob/webpubsub-client/sdk/webpubsub/azure-webpubsub-client/azure/webpubsub/client/_client.py#L67-L69